pkg/sql/conn_executor_exec.go

```diff
@@ -661,8 +662,25 @@ func (ex *connExecutor) commitSQLTransactionInternal(
 	// Now that we've committed, if we modified any table we need to make sure
 	// to release the leases for them so that the schema change can proceed and
 	// we don't block the client.
-	if tables := ex.extraTxnState.tables.getTablesWithNewVersion(); tables != nil {
-		ex.extraTxnState.tables.releaseLeases(ctx)
+	if tables := ex.extraTxnState.tables.GetTablesWithNewVersion(); tables != nil {
+		ex.extraTxnState.tables.ReleaseLeases(ctx)
+	}
+	return nil
+}
+
+// validatePrimaryKeys verifies that all tables modified in the transaction have
+// an enabled primary key after potentially undergoing DROP PRIMARY KEY, which
+// is required to be followed by ADD PRIMARY KEY.
+func validatePrimaryKeys(tc *TableCollection) error {
+	modifiedTables := tc.GetTablesWithNewVersion()
+	for i := range modifiedTables {
+		table := tc.GetUncommittedTableByID(modifiedTables[i].ID).MutableTableDescriptor
+		if !table.HasPrimaryKey() {
+			return unimplemented.NewWithIssuef(48026,
+				"primary key of table %s dropped without subsequent addition of new primary key",
```

Comment:
Is this a new condition? If it is, may be worth isolating in a separate PR beforehand.

Reply:
Correct, that logic felt like it was in the wrong place and given it had a single call site, didn't deserve to be a method on the `TableCollection`.

https://github.com/cockroachdb/cockroach/blob/d7985c53f1e6d32abd604f5987b9a44f6b210ee3/pkg/sql/table.go#L883-L898